src/Common/src/Interop/Windows/Ole32/Interop.CoUnmarshalInterface.cs

```diff
@@ -0,0 +1,19 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.ComTypes;
+
+internal static partial class Interop
+{
+    internal static partial class Ole32
+    {
+        [DllImport(Libraries.Ole32, PreserveSig = false)]
+        internal static extern IntPtr CoUnmarshalInterface(
```

Comment:
You can remove all the ```[In] [Out] attributes``` from interop definations. They are not necessary

Reply:
Guideline is https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/interop-guidelines.md `Use [In] and [Out] only when they differ from the implicit behavior.` 
You are probably right these can go, but they are sometimes important. Automated tests might not not detect if something breaks.

Reply:
I'm not objecting to removing just pointing out they can't automatically be removed in the general case

Reply:
@stephentoub how can we tell when "they differ from the implicit behavior" ? 

Maybe `[In] ref` is an example, because presuambly `ref` is implicitly In/Out?

Reply:
@jkoritzinsky and @AaronRobinsonMSFT can help.

Reply:
In this case the `[In]` can be removed. The reason this is `ref` is for historical reasons with the Win32 C interfaces. If Windows was originally written with a C++ API, then it would have been `const &` instead of `*`. @jkoritzinsky can confirm, but in this case we won't/shouldn't marshal back changes to the `Guid`.

Reply:
I think the meta-point here is that the APIs in question (i.e. Win32) rarely require the attributes due to required stability with a C-style ABI. COM becomes much more complicated historically due to RPC semantics.

Reply:
Yeah the `[In]` isn't needed.

Alternatively, the Guid parameter could be `[MarshalAs(UnmanagedType.LPStruct)] Guid riid`. Because in Win32 the `GUID` parameters are nearly always passed by reference or pointer instead of by value, there's a convenience hook in the marshallers for it.